Update ListenerMakeCommand.php

zanozik opened the pull request:
Allow to make listeners for the events outside of App or Illuminate namespaces, passing event class with absolute path:
```
php artisan make:listener UserCustomListener -e "\Vendor\Package\Events\PackageEvent"
```

zanozik committed:
Update ListenerMakeCommand.php

Allow to make listeners for the events outside of App or Illuminate namespaces

lucasmichot commented on src/Illuminate/Foundation/Console/ListenerMakeCommand.php:
```diff
@@ -54,7 +54,8 @@ protected function buildClass($name)
         if (! Str::startsWith($event, $this->laravel->getNamespace()) &&
-            ! Str::startsWith($event, 'Illuminate')) {
+            ! Str::startsWith($event, 'Illuminate') &&
+            ! Str::startsWith($event, '\\')) {
```
Why calling `Str::startsWith` 3 times? You can call it only once:
```php
$prefixes = [
    $this->laravel->getNamespace(),
    'Illuminate',
    '\\',
];

if (! Str::startsWith($event, $prefixes) {
// ...
```

zanozik replied:
It was called 2 times, so I figured this is a standard practice in Laravel...
Could we call it with anonymous array, to signify that it will not be used anywhere else?
```
if (! Str::startsWith($event, [
    $this->laravel->getNamespace(),
    'Illuminate',
    '\\',
]) {
// ...
```

taylorotwell commented:
So where does this actually place the generated event?

zanozik commented:
@taylorotwell 
`make:listener` only makes a listener for defined event. You thinking `event:generate`, maybe?
In any case, I find it very useful, if I create an event in my package and fire it when a package needs it letting a user create a custom listener by running `php artisan make:listener UserCustomListener -e "\Vendor\Package\Events\MyPackageEvent"`.
I found it strange that we can `make` an outside listener only for Illuminate*, even though there is no restriction on listeners created manually...

zanozik committed:
Update ListenerMakeCommand.php

Pass all prefixes to `Str::startWith()` in anonymous array. instead of calling it 3 times (even more useful if more prefixes will be added later on).

zanozik committed:
Update ListenerMakeCommand.php

Typo :(
